### PR TITLE
refactoring & add JSON support & update convention system

### PR DESCRIPTION
**Pull request by UnLegitCode**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
+
 Lightweight and powerful annotation-based configuration library
 
 ### Usage Guide:
+
 First of all, you need to create a new `FlyconfInstance`.
 `FlyconfInstance` is a specifically configured container for your configurations.
-You can use different Flyconf instances if various configurations use different parsers. It allows you to register custom parsers separately.
+You can use different Flyconf instances if various configurations use different parsers. It allows you to register
+custom parsers separately.
```
Comment (olestxcode) [inline README.md:10]: Please remove this new line you have added

File: build.gradle [hunk 2]
```diff
@@ -13,7 +13,7 @@ allprojects {
             java {
                 toolchain {
                     languageVersion = JavaLanguageVersion.of(javaVersion)
-                    vendor = JvmVendorSpec.matching('Eclipse Adoptium')
+//                    vendor = JvmVendorSpec.matching('Eclipse Adoptium')
```
Comment (olestxcode) [inline build.gradle:16]: Please remove this comment and keep it locally if it doesn't work on your machine

File: json/build.gradle [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+dependencies {
+    api project(':core')
+
+    implementation 'com.google.code.gson:gson:2.10.1'
+}
+
+publishing {
+    publications {
+        yaml(MavenPublication) {
```
Comment (olestxcode) [inline json/build.gradle:16]: Please rename this publication from yaml to json

File: json/.gitignore [hunk 4]
```diff
@@ -0,0 +1,42 @@
+.gradle
```
Comment (olestxcode) [inline json/.gitignore:1]: .gitignore file should not exist in submodule folders
Please move it to the root folder with relevant path changes if needed.

File: yaml/src/main/java/com/github/olestxcode/flyconf/loader/YamlConfigurationLoader.java [hunk 5]
```diff
@@ -1,52 +1,26 @@
...
-
-    public YamlConfigurationLoader(Supplier<InputStream> inputStreamSupplier) {
-        this.inputStreamSupplier = inputStreamSupplier;
+    public YamlConfigurationLoader(@NotNull InputStreamSource streamSource) {
+        super(streamSource);
     }
 
     @Override
-    public Map<String, Object> load() {
+    protected Stream<Map.Entry<String, Object>> load(InputStream stream) {
```
Comment (olestxcode) [inline yaml/src/main/java/com/github/olestxcode/flyconf/loader/YamlConfigurationLoader.java:19]: Won't this level of access affect on method usage outside?

File: build.gradle (hunk 2 above)
Comment (UnLegitCode) [reply build.gradle:16]: This setting is not critical, but may cause problems for users.

File: yaml/src/main/java/com/github/olestxcode/flyconf/loader/YamlConfigurationLoader.java (hunk 5 above)
Comment (UnLegitCode) [reply yaml/src/main/java/com/github/olestxcode/flyconf/loader/YamlConfigurationLoader.java:19]: No

**Pull request merged**